yb-voyager/cmd/exportSchema.go

```diff
@@ -261,29 +261,23 @@ func applyShardedTablesRecommendation(shardedTables []string) (err error) {
 	var newSQLFileContent strings.Builder
 	sqlInfoArr := parseSqlFileForObjectType(filePath, "TABLE")
 	for _, sqlInfo := range sqlInfoArr {
-		var modifiedSqlStmt string
 		/*
 			We can rely on pg_query to detect if it is CreateTable and also table name
 			but due to time constraint this module can't be tested thoroughly so relying on the existing as much as possible
 
 			We can pass the whole .sql file as a string also to pg_query.Parse() all the statements at once.
 			But avoiding that also specially for cases where the SQL syntax can be invalid
 		*/
-		// parsing only the Recommended Sharded Tables
-		if sqlInfo.objName != "" && slices.Contains(shardedTables, sqlInfo.objName) {
-			log.Infof("applying sharding recommendation for table=%q", sqlInfo.objName)
-			modifiedSqlStmt, err = applyShardingRecommendation(&sqlInfo)
-			if err != nil {
-				log.Errorf("failed to apply sharding recommendation for table=%q: %v", sqlInfo.objName, err)
-				fmt.Printf("Unable to apply sharding recommendation for table=%q, continuing without applying...\n", sqlInfo.objName)
-				fmt.Printf("Please manually add the clause \"WITH (colocation = false)\" to the CREATE TABLE DDL of the '%s' table.\n", sqlInfo.objName)
-				modifiedSqlStmt = sqlInfo.formattedStmt
-			} else {
-				log.Infof("original ddl - %s", sqlInfo.stmt)
-				log.Infof("modified ddl - %s", modifiedSqlStmt)
-			}
-		} else {
-			modifiedSqlStmt = sqlInfo.formattedStmt
+		modifiedSqlStmt, match, err := applyShardingRecommendationIfMatching(&sqlInfo, shardedTables)
+		if match && err != nil {
```

Comment:
Just a suggestion on re-organizing the if-else flow: (basically divide the error and non-error cases at the highest level.
```
if err != nil{
    if match {
    } else {
    }
} else {
   if match {}
}
```